plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java

```diff
@@ -365,47 +366,80 @@ private static class StructCoercer
             extends TypeCoercer<RowType, RowType>
     {
         private final List<Optional<TypeCoercer<? extends Type, ? extends Type>>> coercers;
-        private final Block[] nullBlocks;
 
         public StructCoercer(RowType fromType, RowType toType, List<Optional<TypeCoercer<? extends Type, ? extends Type>>> coercers)
         {
             super(fromType, toType);
+            checkArgument(toType.getTypeParameters().size() == coercers.size());
+            checkArgument(fromType.getTypeParameters().size() <= coercers.size());
             this.coercers = ImmutableList.copyOf(requireNonNull(coercers, "coercers is null"));
-            List<Field> toTypeFields = toType.getFields();
-            this.nullBlocks = new Block[toTypeFields.size()];
-            for (int i = 0; i < toTypeFields.size(); i++) {
-                if (i >= fromType.getFields().size()) {
-                    nullBlocks[i] = toTypeFields.get(i).getType().createBlockBuilder(null, 1).appendNull().build();
-                }
-            }
         }
 
         @Override
         public Block apply(Block block)
```

Comment:
This seems to be the same as the code in `DeltaLakeWriter`. Since Delta Lake depends on the Hive code, is there benefit to making this a shared utility method?